### PR TITLE
[security] improve logging/notifications around account security actions

### PR DESCRIPTION
**Pull request by dcramer**
- add general logging behind 'sentry.security' logger
- add mfa-removed event
- add mfa-added event
- add password-changed event
- refactor avatar helpers
- add WIP tag for changesets (Danger)
- add system.security-email option
- add security/support email options to UI

### CODE REVIEW

**Comment by dcramer** [comment]
screenshot 2017-01-16 16 39 51

Remaining TODO:

- [ ] mfa-added event
- [ ] password-changed event

**Comment by dcramer** [comment]
We might want to clean up how events are displayed -- they somewhat look like the screenshot above.

In the future:

- Add regional support to improve upon IP (and potentially hide IP from emails)
- Extend events to include unseen regional login notices
- Add storage for events similar to Activity

File: src/sentry/security/__init__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+def capture_security_activity(account, type, actor, ip_address, context=None,
+                              send_email=True, current_datetime=None):
+    if current_datetime is None:
+        current_datetime = timezone.now()
+
+    logger_context = {
+        'ip_address': ip_address,
+        'user_id': account.id,
+        'actor_id': actor.id,
```
Comment (JTCunning) [inline src/sentry/security/__init__.py]: Would there ever be an instance in which `user_id` and `actor_id` would differ?

Comment (dcramer) [inline src/sentry/security/__init__.py]: @JTCunning when we trigger it as admins (e.g. from _admin)

File: src/sentry/templates/sentry/debug/mail/preview.html [hunk 2]
```diff
@@ -32,6 +32,11 @@
       <optgroup label="Reports">
         <option value="mail/report/">Weekly Report</option>
       </optgroup>
+      <optgroup label="Security">
+        <option value="mail/mfa-added/">MFA Added</option>
+        <option value="mail/mfa-removed/">MFA Removed</option>
+        <option value="mail/mfa-removed/">Password Changed</option>
```
Comment (dcramer) [inline src/sentry/templates/sentry/debug/mail/preview.html]: need to fix this one

File: src/sentry/web/frontend/accounts_twofactor.py [hunk 3]
```diff
@@ -87,12 +112,13 @@ def remove(self, request, interface):
             form = ConfirmPasswordForm(request.user, request.POST)
             if 'password' in form.fields:
                 if form.is_valid():
-                    self.delete_authenticator(interface)
-                    return HttpResponseRedirect(reverse('sentry-account-settings-2fa'))
-                else:
-                    form.errors['__all__'] = ['Invalid password.']
+                    if request.user.check_password(form.cleaned_data['password']):
```
Comment (mattrobenolt) [inline src/sentry/web/frontend/accounts_twofactor.py]: This check is redundant. The form wouldn't be validated if the password was wrong. This happens in English `clean_password` method.

**Review by mattrobenolt — APPROVED** [review]
🔐

File: src/sentry/web/frontend/accounts_twofactor.py (hunk 3 above)
Comment (dcramer) [inline src/sentry/web/frontend/accounts_twofactor.py]: oh I suppose this is my failure at resolving conflicts

**Pull request merged**